elfpy/data/acquire_data.py

```diff
@@ -276,33 +151,32 @@ def main(
                         logging.warning("Error in get_hyperdrive_checkpoint_info, retrying")
                         time.sleep(1)
                         continue
-                if checkpoint_info_dict:  # Proceed only if we have data, otherwise do nothing
-                    # Set defaults
-                    for key in db_schema.CheckpointInfo.__annotations__:
-                        if key not in checkpoint_info_dict:
-                            checkpoint_info_dict[key] = None
-                    # Add to db
-                    postgres.add_checkpoint_infos([db_schema.CheckpointInfo(**checkpoint_info_dict)], session)
+                if checkpoint_info_dict is None:
+                    raise ValueError("Error in getting checkpoint info")
+                block_checkpoint_info = convert_data.convert_checkpoint_info(checkpoint_info_dict)
+                postgres.add_checkpoint_infos([block_checkpoint_info], session)
 
                 # keep querying until it returns to avoid random crashes with ValueError on some intermediate block
                 block_transactions = None
                 for _ in range(RETRY_COUNT):
                     try:
-                        block_transactions = db_schema.fetch_transactions_for_block(
+                        block_transactions = convert_data.fetch_transactions_for_block(
                             web3, hyperdrive_contract, block_number
                         )
                         break
                     except ValueError:
                         logging.warning("Error in fetch_transactions_for_block, retrying")
                         time.sleep(1)
                         continue
-                if block_transactions:  # Proceed only if we have data, otherwise do nothing
-                    postgres.add_transactions(block_transactions, session)
-                if block_transactions and block_pool_info:
-                    wallet_info_for_transactions = get_wallet_info(
-                        hyperdrive_contract, base_contract, block_number, block_transactions, block_pool_info
-                    )
-                    postgres.add_wallet_infos(wallet_info_for_transactions, session)
+
+                if block_transactions is None:  # Proceed only if we have data, otherwise do nothing
+                    raise ValueError("Error in getting transactions")
```

Comment:
I don't know if this warrants an error.  Does convert_data have all the transactions for a block or just the ones we care about.  If we are expecting to see all transactions for a block then have this be a warning 'No transactions in block'

Reply:
The `fetch_transactions_for_block` returns an empty list if there are no transactions in the block, which shouldn't execute this block